storage/src/tests/storageserver/documentapiconvertertest.cpp

```diff
@@ -120,15 +141,46 @@ void DocumentApiConverterTest::testForwardedPut()
     _converter->transferReplyState(*pr, *reply);
 }
 
+void DocumentApiConverterTest::testUpdate()
+{
+    auto update = std::make_shared<document::DocumentUpdate>(_html_type, defaultDocId);
+    documentapi::UpdateDocumentMessage updateMsg(update);
+    updateMsg.setOldTimestamp(1234);
+    updateMsg.setNewTimestamp(5678);
+
+    auto storageCmd = _converter->toStorageAPI(updateMsg, _repo);
+    CPPUNIT_ASSERT_EQUAL(defaultBucket, storageCmd->getBucket());
+
+    auto updateCmd = dynamic_cast<api::UpdateCommand*>(storageCmd.get());
+    CPPUNIT_ASSERT(updateCmd);
+    CPPUNIT_ASSERT_EQUAL(update.get(), updateCmd->getUpdate().get());
+    CPPUNIT_ASSERT_EQUAL(api::Timestamp(1234), updateCmd->getOldTimestamp());
+    CPPUNIT_ASSERT_EQUAL(api::Timestamp(5678), updateCmd->getTimestamp());
+
+    auto mbusReply = updateMsg.createReply();
+    CPPUNIT_ASSERT(mbusReply.get());
+    auto storageReply = _converter->toStorageAPI(static_cast<documentapi::DocumentReply&>(*mbusReply), *storageCmd);
+    auto updateReply = dynamic_cast<api::UpdateReply*>(storageReply.get());
+    CPPUNIT_ASSERT(updateReply);
+
+    auto mbusMsg = _converter->toDocumentAPI(*updateCmd, _repo);
+    auto mbusUpdate = dynamic_cast<documentapi::UpdateDocumentMessage*>(mbusMsg.get());
+    CPPUNIT_ASSERT(mbusUpdate);
+    CPPUNIT_ASSERT((&mbusUpdate->getDocumentUpdate()) == update.get());
+    CPPUNIT_ASSERT_EQUAL(api::Timestamp(1234), mbusUpdate->getOldTimestamp());
+    CPPUNIT_ASSERT_EQUAL(api::Timestamp(5678), mbusUpdate->getNewTimestamp());
+}
```

Comment:
I feel that this test ideally could be broken up into smaller, distinct tests. Currently is a bit hard to follow since it tests all update-related message conversions basically at the same time. Consider refactoring later, not a blocker for this PR.